default.nix

```diff
@@ -0,0 +1,179 @@
+{ stdenvNoCC, writeShellScriptBin, writeText, runCommand,
+  stdenv, fetchurl, makeWrapper, nodejs-10_x, yarn2nix, yarn }:
+with stdenv.lib; let
+  inherit (builtins) fromJSON toJSON split removeAttrs;
+
+  _nodejs = nodejs-10_x;
+  _yarn   = yarn.override { nodejs = _nodejs; };
+
+  depsToFetches = deps: concatMap depToFetch (attrValues deps);
+
+  depFetchOwn = { resolved, integrity, name ? null, ... }:
+    let
+      ssri      = split "-" integrity;
+      hashType  = head ssri;
+      hash      = elemAt ssri 2;
+      bname     = baseNameOf resolved;
+      fname     = if hasSuffix ".tgz" bname || hasSuffix ".tar.gz" bname
+                  then bname else bname + ".tgz";
+    in nameValuePair resolved {
+      inherit name bname;
+      path = fetchurl { name = fname; url = resolved; "${hashType}" = hash; };
+    };
+
+  depToFetch = args @ { resolved ? null, dependencies ? {}, ... }:
+    (optional (resolved != null) (depFetchOwn args)) ++ (depsToFetches dependencies);
+
+  cacheInput = oFile: iFile: writeText oFile (toJSON (listToAttrs (depToFetch iFile)));
+
+  patchShebangs = writeShellScriptBin "patchShebangs.sh" ''
+    set -e
+    source ${stdenvNoCC}/setup
+    patchShebangs "$@"
+  '';
+
+  shellWrap = writeShellScriptBin "npm-shell-wrap.sh" ''
+    set -e
+    if [ ! -e .shebangs_patched ]; then
+      ${patchShebangs}/bin/patchShebangs.sh .
+      touch .shebangs_patched
+    fi
+    exec bash "$@"
+  '';
+
+  npmInfo = src: rec {
+    pkgJson = src + "/package.json";
+    info    = fromJSON (readFile pkgJson);
+    name    = "${info.name}-${info.version}";
+  };
+
+  npmCmd        = "${_nodejs}/bin/npm";
+  npmAlias      = ''npm() { ${npmCmd} "$@" $npmFlags; }'';
+  npmModules    = "${_nodejs}/lib/node_modules/npm/node_modules";
+
+  yarnCmd       = "${_yarn}/bin/yarn";
+  yarnAlias     = ''yarn() { ${yarnCmd} $yarnFlags "$@"; }'';
+
+  npmFlagsYarn  = [ "--offline" "--script-shell=${shellWrap}/bin/npm-shell-wrap.sh" ];
+  npmFlagsNpm   = [ "--cache=./npm-cache" "--nodedir=${_nodejs}" ] + npmFlagsYarn;
+
+  commonEnv = {
+    XDG_CONFIG_DIRS     = ".";
+    NO_UPDATE_NOTIFIER  = true;
+    installJavascript   = true;
+  };
+
+  commonBuildInputs = [ _nodejs makeWrapper ];  # TODO: git?
+
+  # unpack the .tgz into output directory and add npm wrapper
+  # TODO: "cd $out" vs NIX_NPM_BUILDPACKAGE_OUT=$out?
+  untarAndWrap = name: cmds: ''
+    mkdir -p $out/bin
+    tar xzvf ./${name}.tgz -C $out --strip-components=1
+    if [ "$installJavascript" = "1" ]; then
+      cp -R node_modules $out/
+      ${ concatStringsSep ";" (map (cmd:
```

Comment:
I don't know if this is every anything but `npm start --scripts-prepend-node-path=true `.

Reply:
What do you mean by that?

Reply:
ever*. I symlinked `npm` so it could run the `npm start` script, which is somewhat trivial to do (doesn't differ between npm and yarn, afaik)